odoo/addons/base/controllers/avatar.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+
+from odoo.http import Controller, request, route, Response, local_redirect
+from odoo import tools
+
+# show avatar if no image or showAvatar 1
+# show image if image
+
+class Avatar(Controller):
+    @route(['/web/avatar/<int:user_id>'], auth='user', methods=['GET'])
+    def get_avatar(self, user_id=1):
+        if 'isPartner' in request.params and request.params['isPartner'] == '1':
+            user = request.env['res.users'].search([('partner_id', '=', user_id), ('groups_id.id', '=', request.env.ref('base.group_user').id)]).exists()
+        else:
+            user = request.env['res.users'].browse(user_id).exists()
+        if user:
+            if 'consume' in request.params and request.params['consume'] == '1' and user.image_128:
+                return local_redirect(f'/web/image?model=res.users&id={user.id}&field=image_128')
```

Comment:
Why limit it to `image_128`? It's a sensible default but we might need some other size depending on the usage.